pmlc/dialect/pxa/transforms/fusion.cc

```diff
@@ -0,0 +1,434 @@
+// Copyright 2020 Intel Corporation
+
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Affine/IR/AffineValueMap.h"
+#include "pmlc/dialect/pxa/analysis/strides.h"
+#include "pmlc/dialect/pxa/ir/ops.h"
+#include "pmlc/dialect/pxa/transforms/pass_detail.h"
+#include "pmlc/util/logging.h"
+
+namespace pmlc::dialect::pxa {
+
+using mlir::AffineLoadOp;
+using mlir::AffineParallelOp;
+using mlir::Block;
+using mlir::BlockArgument;
+using mlir::Operation;
+using mlir::StrideInfo;
+
+namespace {
+
+using WriteRead = std::pair<AffineReduceOp, AffineLoadOp>;
+using WriteWrite = std::pair<AffineReduceOp, AffineReduceOp>;
+
+struct FusionInfo {
+  // The parallel for ops
+  AffineParallelOp apA;
+  AffineParallelOp apB;
```

Comment:
Naming: could we use `consumer` and `producer`?

Reply:
Producer/consumer sounds too much like it's limited to a single readAfterWrite to me.  Additionally, for most things, we handle the two block similarly, and there are a number of places where I needs someVarA and someVarB, so I'd like to keep the names shortish and regular looking.